### PR TITLE
filter out 0 balances

### PR DESCRIPTION
**Pull request by Jibz-Mysten**
## Description 

Describe the changes or additions included in this PR.
Remove tokens with zero balance.

## Test Plan 
before
https://suiexplorer.com/address/0x94aaaa29f13fddceee955113fe73dd7853586a2a0fdc4863fd998471a666d031
After
https://explorer-git-gj-wallet-filter-zero-balance-mysten-labs.vercel.app/address/0x94aaaa29f13fddceee955113fe73dd7853586a2a0fdc4863fd998471a666d031

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/core/src/hooks/useGetAllBalances.ts [hunk 1]
```diff
@@ -14,7 +15,10 @@ export function useGetAllBalances<TResult = CoinBalance[]>(
     const rpc = useRpcClient();
     return useQuery({
         queryKey: ['get-all-balance', address],
-        queryFn: () => rpc.getAllBalances({ owner: address! }),
+        queryFn: async () => {
+            const balances = await rpc.getAllBalances({ owner: address! });
+            return filterOutZeroBalances(balances);
+        },
```
Comment (mamos-mysten) [inline apps/core/src/hooks/useGetAllBalances.ts:21]: nit : use a `select` here 

also just checking that we want to filter out all coins with zero balance across wallet / explorer

**Review by mamos-mysten — DISMISSED** [review]

File: apps/core/src/hooks/useGetAllBalances.ts (hunk 1 above)
Comment (Jibz-Mysten) [reply apps/core/src/hooks/useGetAllBalances.ts:21]: My understanding is yes, @Jordan-Mysten. This only applies to `getAllBalances`

Comment (mamos-mysten) [reply apps/core/src/hooks/useGetAllBalances.ts:21]: also do we only want to filter out zero balances for unrecognized coins?

Comment (Jibz-Mysten) [reply apps/core/src/hooks/useGetAllBalances.ts:21]: I think we can, but what informational value are coins with zero balance?

**Review by mamos-mysten — APPROVED** [review]

**Pull request merged**